src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs

```diff
@@ -190,5 +202,32 @@ private static HashAlgorithmType MapHashAlgorithmType(Interop.libssl.DataHashAlg
                     return HashAlgorithmType.None;
             }
         }
+        
+        private static int GetHashAlgorithmBitSize(Interop.libssl.DataHashAlgorithm mac)
+        {
+            switch (mac)
+            {
+                case Interop.libssl.DataHashAlgorithm.SSL_MD5:
+                    return MD5_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA1:
+                    return SHA1_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST94:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST89MAC:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA256:
+                    return SHA256_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA384:
+                    return SHA384_HashKeySize;
+
+                default:
+                    return 0;
```

Comment:
@bartonjs : yes, to make it work in veriosn0.9.8 we would have to change the logic for other algorithms too . keyEx and DataCipher.. ..
Btw I don't find  "SSL_CIPHER_get_digest_nid " in 1.0.* existing versions. instead we can use 

#if IMNOTSUREWHAT // OpenSSL 0.9.8
    switch (cipher->algorithms & SSL_MAC_MASK)
    {
        case SSL_MD5:
            cipherMac = NID_md5;
            break;
        case SSL_SHA1:
            cipherMac = NID_sha1;
            break;
        default:
            // Handle this unknown case
            break;
    }
#else
      switch (cipher->algorithms_mac)
    {
      //do mapping the way it is done currently.
    }
#endif
..